data/extension/init/templates/tinygo/envoy.filters.network/default/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+var (
+	connectionCounterName = "proxy_wasm_go.connection_counter"
+	counter               proxywasm.MetricCounter
+)
+
+type context struct{ proxywasm.DefaultContext }
+
+func main() {
+	proxywasm.SetNewRootContext(func(contextID uint32) proxywasm.RootContext { return context{} })
+	proxywasm.SetNewStreamContext(func(contextID uint32) proxywasm.StreamContext { return context{} })
+}
+
+func (ctx context) OnVMStart(int) bool {
+	var err error
+	counter, err = proxywasm.DefineCounterMetric(connectionCounterName)
+	if err != nil {
+		proxywasm.LogCritical("failed to initialize connection counter: ", err.Error())
+	}
+	return true
+}
+
+func (ctx context) OnNewConnection() types.Action {
+	proxywasm.LogInfo("new connection!")
+	return types.ActionContinue
+}
+
+func (ctx context) OnDone() bool {
```

Comment:
What effect would returning `false` have ?

Reply:
this is httpContext so there's no effect IIUC but the current implementation is just to share OnDone interface among Root/Netwrok/Http contexts. To be honest, I'm a little bit confused and I think the current interface is problematic just like the comment https://github.com/proxy-wasm/spec/issues/9#issuecomment-681177169

will do the overhaul of the interface